src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleDescriptorTest.cs

```diff
@@ -8,70 +10,56 @@ namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules
+            descriptor.CanSkipValidation.Should().BeFalse();
         }
 
         [Fact]
-        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_CanSkipValidation()
         {
-            var rule = new TestConsensusRuleWithOptionalValidationAttribute("myRule");
-
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithOptionalValidationAttribute());
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithOptionalValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithOptionalValidationAttribute).RuleName);
+            descriptor.Rule.Should().BeOfType<RuleWithOptionalValidationAttribute>();
+            descriptor.RuleAttributes.Should().HaveCount(1);
+            descriptor.RuleAttributes.Single().Should().BeOfType<ValidationRuleAttribute>();
 
-            Assert.Equal(3, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
-            Assert.True(descriptor.Attributes[2] is ValidationRuleAttribute);
-
-            Assert.True(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeTrue();
         }
 
         [Fact]
-        public void Constructor_ConsensusRuleWithoutValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithoutValidationAttributeButWithOtherRuleAttribute_CannotSkipValidation()
         {
-            var rule = new TestConsensusRuleWithoutValidationAttribute("myRule");
-
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithMempoolAttributeButWithoutValidationAttribute());
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithoutValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithoutValidationAttribute).RuleName);
+            descriptor.Rule.Should().BeOfType<RuleWithMempoolAttributeButWithoutValidationAttribute>();
 
-            Assert.Equal(2, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
+            descriptor.RuleAttributes.Should().HaveCount(1);
 
-            Assert.False(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeFalse();
         }
 
-        [MempoolRule]
-        [ExecutionRule]
-        [ValidationRule(CanSkipValidation = false)]
-        private class TestConsensusRuleWithValidationAttribute : ConsensusRule
+        [Fact]
+        public void Constructor_ConsensusRuleWithZeroRuleAttributes_CanSkipValidation()
```

Comment:
also added this additional test where there are no attributes at all, as behaviour is different

Reply:
the diff is a bit screwy

Reply:
yep, good stuff.